internal/databases/postgres/catchup_send_recieve_handler.go

```diff
@@ -0,0 +1,289 @@
+	tracelog.ErrorLogger.FatalOnError(err)
+	tracelog.InfoLogger.Printf("Send done")
+}
+
+func sendFileCommands(encoder *gob.Encoder, directory string, list internal.BackupFileList, checkpoint LSN) {
+	extendExcludedFiles()
+	err := filepath.Walk(directory, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			if os.IsNotExist(err) {
+				tracelog.WarningLogger.Println(path, " deleted during filepath walk")
+				return nil
+			}
+			return err
+		}
+		if info.Name() == PgControl {
+			return nil
+		}
+		fileName := info.Name()
+		_, excluded := ExcludedFilenames[fileName]
+		isDir := info.IsDir()
+		if isDir && excluded {
+			return filepath.SkipDir
+		}
+		if isDir {
+			return nil
+		}
+		if excluded {
+			return nil
+		}
+		fullFileName := utility.GetSubdirectoryRelativePath(path, directory)
+
+		wasInBase := false
+		if fdto, ok := list[fullFileName]; ok {
+			if fdto.MTime.Equal(info.ModTime()) {
+				// No need to catchup
+				return nil
+			}
+			wasInBase = true
+		}
+
+		increment := isPagedFile(info, path) && wasInBase
+
+		var fd io.ReadCloser
+		var size int64
+		if !increment {
+			fd, err = os.Open(path)
+			tracelog.ErrorLogger.FatalOnError(err)
+			size = info.Size()
+		} else {
+			fd, size, err = ReadIncrementalFile(path, info.Size(), checkpoint, nil)
+
+			if _, ok := err.(*InvalidBlockError); ok {
+				fd, err = os.Open(path)
+				tracelog.ErrorLogger.FatalOnError(err)
+				size = info.Size()
+				increment = false
+			} else {
+				tracelog.ErrorLogger.FatalOnError(err)
+			}
+		}
+
+		//size := info.Size()
+		err = encoder.Encode(CatchupCommandDto{FileName: fullFileName, IsFull: !increment, FileSize: uint64(size), IsIncremental: increment})
+		tracelog.ErrorLogger.FatalOnError(err)
+		reader := io.MultiReader(fd, &ioextensions.ZeroReader{})
+
+		for size != 0 {
+			min := 8192
+			if int64(min) > size {
+				min = int(size)
+			}
+			var bytes = make([]byte, min)
+			_, err := io.ReadFull(reader, bytes)
+			tracelog.ErrorLogger.FatalOnError(err)
+			size -= int64(len(bytes))
+			err = encoder.Encode(bytes)
+			tracelog.ErrorLogger.FatalOnError(err)
+		}
+
+		tracelog.InfoLogger.Printf("Sent %v, %v bytes", fullFileName, info.Size())
+		tracelog.ErrorLogger.FatalOnError(err)
+		err = fd.Close()
+		tracelog.ErrorLogger.FatalOnError(err)
+
+		return nil
+	})
+	tracelog.ErrorLogger.FatalOnError(err)
+	tracelog.DebugLogger.Printf("Filepath walk done")
+}
+
+func HandleCatchupReceive(pgDataDirectory string, port int) {
+	pgDataDirectory = utility.ResolveSymlink(pgDataDirectory)
+	tracelog.InfoLogger.Printf("Receiving %v on port %v\n", pgDataDirectory, port)
+	listen, err := net.Listen("tcp", fmt.Sprintf(":%v", port))
+	tracelog.ErrorLogger.FatalOnError(err)
+	conn, err := listen.Accept()
+	sendControlAndFileList(pgDataDirectory, err, conn)
+	decoder := gob.NewDecoder(conn)
+	for {
+		var cmd CatchupCommandDto
+		err := decoder.Decode(&cmd)
+		if io.EOF == err {
+			break
```

Comment:
I would expect to see `Unexpected EOF` error here, rather than silent exit with following `Receive done` logging.

Reply:
Ugh... that was a debug leftover. Thanks for spotting the bug!